src/libsystemd/sd-id128/sd-id128.c

```diff
@@ -148,6 +148,9 @@ _public_ int sd_id128_get_boot(sd_id128_t *ret) {
                 r = id128_read("/proc/sys/kernel/random/boot_id", ID128_UUID, &saved_boot_id);
                 if (r < 0)
                         return r;
+
+                if (sd_id128_is_null(saved_machine_id))
```

Comment:
hmm, if the boot ID is all zeroes things are really weird ;-)

(but maybe this is OK to keep, in case people do weird stuff such as having fake /proc/ in disk files for testing purposes or so)